src/_pytest/assertion/util.py

```diff
@@ -520,7 +520,7 @@ def _compare_eq_dict(
                 + " != "
                 + highlighter(saferepr({k: right[k]}))
             ]
-    extra_left = set_left - set_right
+    extra_left = [k for k in left if k not in right]
```

Comment:
It is more efficient to reuse the set for the containment check:


```suggestion
    extra_left = [k for k in left if k not in set_right]
```